charts/opentelemetry-operator/templates/_helpers.tpl

```diff
@@ -19,3 +19,14 @@ Selector labels
 app.kubernetes.io/name: {{ include "opentelemetry-operator.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
+
+
+{{/*
+Manager add environment variables from map to env obj.
+*/}}
+{{- define "opentelemetry-operator.env" -}}
+{{- range $name, $value := .Values.manager.env }}
```

Comment:
if the values.yaml env section contains entries like:
`- testing: a-value`
it will create name value entries that look something like:
```
- name: 0
  value: "map[testing:a-value]"
```

Thats definitely user error, but it'll pass check and install ok which might not be a great user experience.  Is there anything we can do to enforce a map in the values.yaml?

Reply:
@dmitryax another use case for expanding the values.schema.json for operator charts?

Reply:
@TylerHelmuth We can try do something like:

```
{{/*
Manager add environment variables from map to env obj.
*/}}
{{- define "opentelemetry-operator.env" -}}
{{ $envs := .Values.manager.env }}
{{- range $name, $value := .Values.manager.env }}
{{- if kindIs "int" $name }}
{{ fail "Value for .Values.manager.env is invalid. .Values.manager.env is a map of "KEY: value" pairs e.g. KEY: value" }}
{{ else }}
- name: {{ $name }}
  value: {{ $value | quote -}}
{{- end }}
{{- end }}
{{- end }}
```

In case of `- key: value` it will return execution error with message.

Reply:
That would work but I like the simplicity of your original solution. I think we can probably enforce it with the schema and until that's in place there is your example in the values.yaml.